Sources/SourceKitLSP/DocumentManager.swift

```diff
@@ -140,27 +141,25 @@ public final class DocumentManager {
   ///   - newVersion: The new version of the document. Must be greater than the
   ///     latest version of the document.
   ///   - edits: The edits to apply to the document
-  ///   - willEditDocument: Optional closure to call before each edit. Will be
-  ///     called multiple times if there are multiple edits.
+  ///   - requiresUpcomingEdits: Flag to return upcoming editst. Will be
+  ///     calling ``willEditDocument`` for each edit made
   /// - Returns: The snapshot of the document before the edit and the snapshot
-  ///   of the document after the edit.
+  ///   of the document after the edit, including all edits that were made.
   @discardableResult
   public func edit(
     _ uri: DocumentURI,
     newVersion: Int,
-    edits: [TextDocumentContentChangeEvent],
-    willEditDocument: ((_ before: LineTable, TextDocumentContentChangeEvent) -> Void)? = nil
-  ) throws -> (preEditSnapshot: DocumentSnapshot, postEditSnapshot: DocumentSnapshot) {
+    edits: [TextDocumentContentChangeEvent]
+  ) throws -> (preEditSnapshot: DocumentSnapshot, postEditSnapshot: DocumentSnapshot, edits: [SourceEdit]) {
     return try queue.sync {
       guard let document = documents[uri] else {
         throw Error.missingDocument(uri)
       }
       let preEditSnapshot = document.latestSnapshot
 
+      var upcomingEdits: [SourceEdit] = []
```

Comment:
I’m not a huge fan of the `upcomingEdits` naming. Maybe `sourceEdits` is a better name? At least that describes that the differentiator between this array and `edits` is that it has a different type of edit.